data-structures-queues/take-front.js

```diff
@@ -0,0 +1,8 @@
+export default function takeFront(queue) {
+  const x = queue.peek();
+  if (x === undefined) {
+    return undefined;
+  } else {
+    return queue.dequeue();
+  }
```

Comment:
Can't you just return `queue.dequeue()` and forget the rest? `dequeue` returns `undefined` if the queue is empty.